book/workflow.rst

```diff
@@ -0,0 +1,245 @@
+be applied to and the initial step that is applied when initializing the workflow for an entity.
+
+Defining the Workflow Steps
+...........................
+
+Next, you need to define the states an entity can have when the workflow is applied:
+
+.. code-block:: yaml
+    :linenos:
+
+    workflows:
+        phone_call:
+            # ...
+            steps:
+                start_call:
+                    label: 'Start Phone Call'
+                    allowed_transitions:
+                        - connected
+                        - not_answered
+                start_conversation:
+                    label: 'Call Phone Conversation'
+                    allowed_transitions:
+                        - end_conversation
+                end_call:
+                    label: 'End Phone Call'
+                    is_final: true
+
+In this example, the entity can have three states: ``start_call`` (which was configured as the
+start step above), ``start_conversation`` and ``end_call``. The ``end_call`` step is marked as
+final which means that the workflow terminates when this step is reached. The ``final`` flag does
+not mean that there cannot be any further transitions, but it is used internally for additional
+data manipulation (for example, report entities whose workflow is in a final step).
+
+The ``allowed_transitions`` defines which :ref:`transitions <book-workflow-transitions>` can be
+applied in a step.
+
+Workflow Attributes
+...................
+
+During the transition from one step to another step, the user can modify attributes of the
+underlying entity. All attributes that can be modified in any of the transitions must be enabled
+with the ``attributes`` option:
+
+.. code-block:: yaml
+    :linenos:
+
+    workflows:
+        phone_call:
+            # ...
+            attributes:
+                phone_call:
+                    label: Phone Call
+                    type: entity
+                    options:
+                        class: Acme\Bundle\DemoWorkflowBundle\Entity\PhoneCall
+                call_timeout:
+                    type: integer
+                    label: 'Call Timeout'
+                call_successfull:
+                    type: boolean
+                    label: 'Call Successful'
+                conversation_successful:
+                    type: boolean
+                    label: 'Conversation Successful'
+                conversation_comment:
+                    type: string
+                    label: 'Conversation Comment'
+                conversation_result:
+                    type: string
+                    label: 'Conversation Result'
+                conversation:
+                    type: entity
+                    label: 'Conversation'
+                    options:
+                        class: Acme\Bundle\DemoWorkflowBundle\Entity\PhoneConversation
+
+.. tip::
+
+    This example assumes that the keys used under the ``attributes`` option are the names of the
```

Comment:
I'd say that's not correct. In fact attribute names have no restrictions - attribute name is just a string identifier used to access (and modify) data. 

There are two ways to store this data:
 - directly inside workflow item, full attribute definition required (label, type etc.) - it means that attribute data will not be automatically copied to related entity (or extracted from related entity), so it can be accessible only in scope of this workflow item; this case is useful if we need to store some temporary data, or data that make sense only in scope of specific workflow item
 - inside related entity, property path required - it means that attribute data will be automatically copied to related entity (or extracted from related entity) by specified property paths, in other words such attributes are just the links to specified value; this case is useful if we need to edit entity fields.

Usually attribute names are similar to entity properties names, but this is not a rule - it's just a convenient way to describe workflow in a readable form.